### DISCUSSION TITLE
Optional weaker propagation for NoOverlap constraints

### DISCUSSION THREAD

**Discussion by ozan-cb** [google/or-tools#3267] [Feature requests] [answered]
**What language and solver does this apply to?**
All / CP-SAT

**Describe the problem you are trying to solve.**
I am working on a scheduling problem that makes use of thousands of binary NoOverlap constraints over thousands of time units, and some additional arithmetic constraints. The number of conflicts reported by the solver is usually close to 0 regardless of the number of intervals and constraints, however as I increase the size of the problem the runtime of the solver increases much faster than I'd like.

This hints to me that propagation of NoOverlap constraints are heavier than I need, and it seems like there are multiple propagators that are registered within the solver:
https://github.com/google/or-tools/blob/49b6301e1e1e231d654d79b6032e79809868a70e/ortools/sat/disjunctive.cc#L88

When I remove the not-last and edge finding propagators, recompile, and run again I get more than 4x speedup. Alternatively, when I remove only edge finding I get about 3x speedup. In either alternative, the number of conflicts do not change much.

**Describe the solution you'd like**

Addition of two fields to `SatParameters` for optionally disabling these propagators would be very beneficial for my use case.

**Describe alternatives you've considered**

- Decomposition of NoOverlap constraints into logical+linear constraints creates too many clauses, and takes even more time than the default runtime. 
- `SatParameters` fields that are relevant do not impact runtime (e.g., `use_precedences_in_disjunctive_constraint`)

**Additional context**
I'd be happy to do the implementation myself if this sounds like a good addition to CP-SAT.

**Comment by lperron** [comment]
Can you try setting the `use_combined_no_overlap` parameter to true ?

**Reply by ozan-cb** [reply]
`use_combined_no_overlap` was one of the flags I tried toggling before, since that seems to replace the multiple propagators I mentioned above. Unfortunately, with that flag runtime of the solver more than doubles in at least two instances that I have.

**Comment by lperron** [comment, marked as answer]
can you send us the proto of the model ?

**Reply by ozan-cb** [reply]
Apologies for the late reply. I realized that if I use v9.3 instead of v9.0 toggling the mentioned propagators doesn't lead to the significant runtime difference I saw before, and the CP-SAT solver by default about 2x faster on the benchmarks I have. 

I think being able to tweak the propagation strength for global constraints is still a nice optional feature to have, but I'm okay with closing this feature request.
